CMakeLists.txt

```diff
@@ -38,9 +38,17 @@ if(PANDORA_MONITORING)
     add_definitions("-DMONITORING")
 endif()
 
+# Set up C++ Standard
+# ``-DCMAKE_CXX_STANDARD=<standard>`` when invoking CMake
+set(CMAKE_CXX_STANDARD 11 CACHE STRING "")
+
+if(NOT CMAKE_CXX_STANDARD MATCHES "11|14|17|20")
+  message(FATAL_ERROR "Unsupported C++ standard: ${CMAKE_CXX_STANDARD}")
+endif()
```

Comment:
Thanks for the proposed changes Thomas. My main request here would be to avoid this restriction. We only want to require a minimum standard (at which point the previous set of CMAKE_CXX_STANDARD should be sufficient), not apply future constraints. (We should now be able to make this minimum 17 rather than 11).